app/lib/shared/markdown.dart

```diff
@@ -400,3 +400,86 @@ Version? _extractVersion(String? text) {
     return null;
   }
 }
+
+class ScreenshotImage {
+  final String url;
+  final String? title;
+
+  ScreenshotImage(this.url, {this.title});
+
+  @visibleForTesting
+  Map<String, dynamic> toJson() => {
+        'url': url,
+        if (title != null) 'title': title,
+      };
+}
+
+@visibleForTesting
+List<ScreenshotImage> extractScreenshotImages(String markdownSource) {
+  final visitor = _ScreenshotImageVisitor();
+  final nodes = _parseMarkdownSource(markdownSource, false);
+  nodes.forEach((node) => node.accept(visitor));
+  return visitor.images;
+}
+
+/// Finds screenshot images.
+///
+/// When images are embedded in markdown, we could say that it is a screenshot image, if:
+/// - (1) the image is also a link to the image, or
+/// - (2) the image has an alt-text that includes the word "screenshot", or
+/// - (3) the image has a custom data-property only recognized by pub.dev.
```

Comment:
We should only do one of these right?

Reply:
We can do (1) and (2) easily with basic markdown support, and I think it is not controversial to support both. I think that (3) is more involved with markdown pruning the HTML markup's output, so I'd happily drop that for now.